File: mobile/src/types/deal.ts
export type DealStatus =
  | 'draft'
  | 'company_enriched'
  | 'quote_added'
  | 'indicative_offer_ready'
  | 'submitted'
  | 'internal_review'
  | 'missing_documents'
  | 'pre_approved'
  | 'refi_package_ready'
  | 'refi_review'
  | 'financier_approved'
  | 'financier_rejected'
  | 'firm_offer_generated'
  | 'contract_generated'
  | 'signing'
  | 'signed'
  | 'activation_pending'
  | 'active'
  | 'cancelled'

export interface Deal {
  id: string
  public_id: string
  company_id: string
  partner_org_id: string | null
  submitted_by_user_id: string | null
  status: DealStatus
  amount_cents: number | null
  currency: string
  duration_months: number | null
  risk_score: number | null
  risk_band: 'green' | 'orange' | 'red' | null
  monthly_payment_cents: number | null
  created_at: string
  updated_at: string
}

export interface DealCreatePayload {
  company_id: string
  amount_cents?: number | null
  duration_months?: number | null
  currency?: string
}

export interface DealStatusTransitionPayload {
  status: DealStatus
}

File: backend/app/schemas/deal.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class DealCreateRequest(BaseModel):
    company_id: uuid.UUID
    amount_cents: int | None = None
    currency: str = "EUR"
    duration_months: int | None = None


class DealPatchRequest(BaseModel):
    amount_cents: int | None = None
    duration_months: int | None = None
    currency: str | None = None


class DealStatusRequest(BaseModel):
    status: str


class DealResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    public_id: str
    company_id: uuid.UUID
    partner_org_id: uuid.UUID | None
    submitted_by_user_id: uuid.UUID | None
    status: str
    amount_cents: int | None
    currency: str
    duration_months: int | None
    risk_score: float | None
    risk_band: str | None
    monthly_payment_cents: int | None
    created_at: datetime
    updated_at: datetime
